src/serverenvironment.cpp

```diff
@@ -800,11 +800,29 @@ class ABMHandler
 		return active_object_count;
 
 	}
-	void apply(MapBlock *block)
+	void apply(MapBlock *block, int &abm_checked, int &abm_run, int &abm_not_cached)
 	{
 		if(m_aabms.empty() || block->isDummy())
 			return;
 
+		if (block->contents_cached) {
+			bool run_abms = false;
+			for (content_t c : block->contents) {
```

Comment:
this could use a comment to easily see what it does

Reply:
Yep!